<a href="https://colab.research.google.com/github/rpizarrog/IA/blob/main/book%20notes/Cargar_insurance_local.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Objetivo
Cargar y explorar el archivo insurance.csv desde una ruta o directorio de una computadora local.

# Descripción 

El archivo fue descargado a equipo local desde el enlace: [kaggle.com](https://www.kaggle.com/datasets/mirichoi0218/insurance?resource=download). https://www.kaggle.com/datasets/mirichoi0218/insurance?resource=download

El conjunto de datos son observaciones relacionadas con variables de interés para una aseguradora en relación a seguros médicos de personas. 

Las variables son:

age: edad del beneficiario 

sex: género del contratante female o male. (Femenino o Masculino)

bmi: Indice de masa corporal que es la relación entre peso y estatura. Dividiendo los kilogramos de peso por el cuadrado de la estatura en metros (IMC = peso (kg)/ estatura (m) ^ 2.
$$
IMC = \frac{peso}{estatura^{2}}
$$


children: Número de hijos o dependientes.

smoker: Si fuma o no fuma.

region: Área de residencias en EStados Unidos (northeast, southeast, southwest, northwest).

charges: El costos de la aseguranza por servicios médicos.

## Cargar archivos 


In [5]:
# Cargar archivos
from google.colab import files
archivo = files.upload()

Saving insurance.csv to insurance (1).csv


In [11]:
# Pandas
import pandas as pd
import io

datos = pd.read_csv( io.BytesIO( archivo['insurance.csv']))




## Análisis exploratorio

In [12]:
# Explorando los datos
datos
print("Son 1338 observaciones 7 variables o columnas")

Son 1338 observaciones 7 variables o columnas


In [13]:
# Los primeros registros con head()
datos.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [14]:
# Los últimos registros con head()
datos.tail()

age     sex    bmi  children smoker     region     charges
1333   50    male  30.97         3     no  northwest  10600.5483
1334   18  female  31.92         0     no  northeast   2205.9808
1335   18  female  36.85         0     no  southeast   1629.8335
1336   21  female  25.80         0     no  southwest   2007.9450
1337   61  female  29.07         0    yes  northwest  29141.3603

In [15]:
# Estadísticos descriptivos
datos.describe().transpose()

count          mean           std        min         25%       50%  \
age       1338.0     39.207025     14.049960    18.0000    27.00000    39.000   
bmi       1338.0     30.663397      6.098187    15.9600    26.29625    30.400   
children  1338.0      1.094918      1.205493     0.0000     0.00000     1.000   
charges   1338.0  13270.422265  12110.011237  1121.8739  4740.28715  9382.033   

                   75%          max  
age          51.000000     64.00000  
bmi          34.693750     53.13000  
children      2.000000      5.00000  
charges   16639.912515  63770.42801

De las variables del conjunto de datos hay 4 variables con tipos de datos numéricos; age, bmi children charges. 

La media de la edad del asegurado es 39 años con una desviación estándar de 14 años, la edad mínima es 18 y máxima es 64 años.

Con respecto al índice de masa corporal (IMC) del asegurado la media aritmética es de 30.66 con una desviacion estándra de 6.09. El valor mínimo es de 15.96 y el valor máximo es de 53.13.

El número de hijos está entre 1 y 5 con una media aritmética de 1.09 y una desviación estándra de 1.02.

La media artimética de los costos por aseguranza médica es de 13270.42, con una desviación estándar de 12110.01. Los valores mínimo y máximo están en el intervalo de 1121.87 a	63770.42.


## La estructura de los datos


In [17]:
datos.shape


(1338, 7)

Son 1338 observaciones 7 variables o columnas

## Verificar valores nulos

In [20]:
datos.isnull().sum()


age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

No existen valores nulos en el conjunto de datos

## Los nombres de las columnas


In [18]:
datos.columns

Index(['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

## Columnas numéricas y categóricas



In [28]:
datos.dtypes

age           int64
sex          object
bmi         float64
children      int64
smoker       object
region       object
charges     float64
dtype: object

age, bmi, children y charges son numéricas.

sex, smoker y region son variables categóricas